syft/frameworks/torch/tensors/interpreters/replicative_shared.py

```diff
@@ -0,0 +1,154 @@
+import torch
+from syft.generic.abstract.tensor import AbstractTensor
+import random
+import syft
+from operator import add, sub
+
+
+class ReplicatedSharingTensor(AbstractTensor):
+    def __init__(
+        self, shares_map=None, owner=None, id=None, tags=None, description=None,
+    ):
+        super().__init__(id=id, owner=owner, tags=tags, description=description)
+        self.child = shares_map
+        self.ring_size = 2 ** 5
+
+    def share_secret(self, secret, workers):
+        number_of_shares = len(workers)
+        workers = self.__arrange_workers(workers)
+        shares = self.generate_shares(secret, number_of_shares)
+        shares_map = self.__distribute_shares(workers, shares)
+        self.child = shares_map
+        return self
+
+    @staticmethod
+    def __arrange_workers(workers):
+        """ having local worker in index 0 saves one communication round"""
+        if syft.hook.local_worker in workers:
+            workers.pop(syft.hook.local_worker)
+            workers = [syft.hook.local_worker] + workers
+        return workers
+
+    def generate_shares(self, plain_text, number_of_shares=3):
+        shares = []
+        for _ in range(number_of_shares - 1):
+            shares.append(torch.tensor(random.randrange(self.ring_size)))
+        shares.append(torch.tensor((plain_text - sum(shares)) % self.ring_size))
+        return shares
+
+    @staticmethod
+    def __distribute_shares(workers, shares):
+        shares_map = {}
+        for i in range(len(shares)):
+            pointer1 = shares[i].send(workers[i])
+            pointer2 = shares[(i + 1) % len(shares)].send(workers[i])
+            shares_map[workers[i]] = (pointer1, pointer2)
+        return shares_map
+
+    def reconstruct(self):
+        shares_map = self.get_shares_map()
+        shares = self.__retrieve_shares(shares_map)
+        plain_text_mod = self.__sum_shares(shares)
+        plain_text = self.__map_modular_to_real(plain_text_mod)
+        return plain_text
+
+    def __retrieve_shares(self, shares_map):
+        pointers = self.__retrieve_pointers(shares_map)
+        shares = []
+        for pointer in pointers:
+            shares.append(pointer.get())
+        return shares
+
+    @staticmethod
+    def __retrieve_pointers(shares_map):
+        players = list(shares_map.keys())
+        pointers = list(shares_map[players[0]])
+        pointers.append(shares_map[players[1]][1])
+        return pointers
+
+    def __sum_shares(self, shares):
+        return sum(shares) % self.ring_size
+
+    def __map_modular_to_real(self, mod_number):
+        """In a modular ring, a number x is mapped to a negative
+         real number ]0,-∞[ iff x > ring_size/2 """
```

Comment:
TYPO: ```[``` should be ```]``` and ```]``` should be ```[```

Reply:
that's not actually a typo, open bracket means the set doesn't include that number. negative numbers are everything between 0 and - infinity not including them, I can remove it if it's confusing, "negative real number" is kinda sufficient